src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx

```diff
@@ -0,0 +1,74 @@
+import { memo, useEffect, useState } from 'react';
+import { deleteDatabaseProfileAvatar, getDatabaseProfileAvatar } from '../../../../../drive/services/database.service';
+import Avatar from '../../../../../shared/components/Avatar';
+import { useAppDispatch } from '../../../../../store/hooks';
+import { updateUserAvatarThunk } from '../../../../../store/slices/user';
+
+export const extractExpiresValue = (url: string): number | null => {
+  const regex = /Expires=(\d+)/;
+  const match = url.match(regex);
+  return match ? parseInt(match[1]) : null;
+};
+
+export const extractAvatarURLID = (url: string): string | null => {
```

Comment:
Remove this function if not required

Reply:
is used to extract the uuid from the avatar url and compare it with the uuid in the cache.

Reply:
But, the avatar UUID is in `xUser.avatar`, or at least, it should be

Reply:
Ah, okay, the xUser.avatar has the presigned-url, not just the UUID, okay then